### ISSUE TITLE
padded-blocks fails without semicolons

### ISSUE THREAD

**Issue by jdelStrother** [eslint/eslint#2273]
I think this ought to give me a "Block must not be padded by blank lines" warning : 

```js
/* globals window */
/* eslint padded-blocks: [2, "never"], semi: [2, "never"] */
if (window.foo) {
  window.bar = 1

}
```

It seems that padded blocks only works in the presence of semicolons - eg 

```js
/* globals window */
/* eslint padded-blocks: [2, "never"], semi: [2, "always"] */
if (window.foo) {
  window.bar = 1;

}
```
will correctly give me the padded-blocks warning.

(Sort of reluctant to post this issue - I like being able to split up if/else blocks with a trailing newline in the `if` block...)

**Comment by btmills** [comment]
ASI inserts the semicolon immediately before the `}` in the first example, so the assignment statement technically spans the empty line according to the parser. `padded-blocks` should probably start checking the location of the last actual token rather than relying on the implicit semicolon from the parser.

**Comment by btmills** [comment]
Working on this.

**Comment by Flet** [comment]
Should a comment be considered a blank line?

We're seeing this comeback as a padded-blocks error with 0.19.0:
```js
    if (argv.airplay) {
      var airplay = require('airplay-js')
      airplay.createBrowser()
        .on('deviceOn', function (device) {
          device.play(href, 0, function () {})
        })
        .start()
      // TODO: handle case where user closes airplay. do same thing as when VLC is closed
    }
```

see this issue for more details: feross/standard#124
